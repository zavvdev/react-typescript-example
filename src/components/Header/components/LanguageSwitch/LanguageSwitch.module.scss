@import "src/styles/theme";

.root {
  display: flex;
  align-items: center;

  .langWrap {
    & span {
      cursor: pointer;
    }

    .lang {
      cursor: pointer;
      border: none;
      background: none;
      text-transform: uppercase;
      padding: 10px;

      &.active {
        color: $cPrimary;
        font-weight: bold;
      }
    }
  }
}