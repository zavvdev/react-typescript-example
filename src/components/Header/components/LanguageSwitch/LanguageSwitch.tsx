import cn from 'classnames';
import { useTranslation } from 'react-i18next';
import { APP_LANGUAGES } from 'src/constants';
import classes from 'src/components/Header/components/LanguageSwitch/LanguageSwitch.module.scss';

const LanguageSwitch = () => {
  const { i18n } = useTranslation();
  const languages = Object.values(APP_LANGUAGES);

  const handleLanguageChange = (nextLang: string) => {
    i18n.changeLanguage(nextLang);
  };

  return (
    <div className={classes.root}>
      {languages.map((lang, index) => (
        <div className={classes.langWrap} key={lang}>
          <button
            type="button"
            onClick={() => handleLanguageChange(lang)}
            className={cn(classes.lang, {
              [classes.active]: i18n.language === lang,
            })}
          >
            {lang}
          </button>
          {index === 0 ? <span>|</span> : null}
        </div>
      ))}
    </div>
  );
};

export default LanguageSwitch;
