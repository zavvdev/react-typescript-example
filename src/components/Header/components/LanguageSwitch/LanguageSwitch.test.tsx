import { screen, fireEvent } from '@testing-library/react';
import { mockI18nChangeLanguage } from 'src/tests/mocks/node-modules/react-i18next';
import render from 'src/tests/utils/render';
import LanguageSwitch from 'src/components/Header/components/LanguageSwitch/LanguageSwitch';

describe('<LanguageSwitch />', () => {
  test('should render LanguageSwitch component (snapshot)', () => {
    const { asFragment } = render(<LanguageSwitch />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should call i18n changeLanguage function', () => {
    render(<LanguageSwitch />);
    const enBtn = screen.getByText('en');
    const ruBtn = screen.getByText('ru');
    fireEvent.click(ruBtn);
    expect(mockI18nChangeLanguage).toBeCalledWith('ru');
    fireEvent.click(enBtn);
    expect(mockI18nChangeLanguage).toBeCalledWith('en');
  });
});
