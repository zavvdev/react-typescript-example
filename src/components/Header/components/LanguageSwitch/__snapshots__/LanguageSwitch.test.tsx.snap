// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSwitch /> should render LanguageSwitch component (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="langWrap"
    >
      <button
        class="lang active"
        type="button"
      >
        en
      </button>
      <span>
        |
      </span>
    </div>
    <div
      class="langWrap"
    >
      <button
        class="lang"
        type="button"
      >
        ru
      </button>
    </div>
  </div>
</DocumentFragment>
`;
